### PR TITLE
refactor: message struct

### PR DESCRIPTION
**Pull request by soulbird**
Refactor message: Using structured data structures to pass messages

### CODE REVIEW

File: internal/core/message/a6conf.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		All: make(map[string]interface{}),
+	}
+	err := unMarshal(value, a6)
+	if err != nil {
+		return nil, err
+	}
+	return a6, nil
+}
+
+func unMarshal(data []byte, v *A6Conf) error {
```
Comment (spacewander) [inline internal/core/message/a6conf.go]: Use unmarshal would be better?

File: internal/core/message/message.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package message
+
+type StoreEvent = int
+
+const (
+	// 配置更新
```
Comment (spacewander) [inline internal/core/message/message.go]: Please use English

File: internal/core/message/message.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	return msg, nil
+}
+
+func (msg *Message) ServiceName() string {
+	up := msg.a6Conf.Upstream
+	if up.ServiceName != "" {
+		return up.ServiceName
+	}
+	return up.DupServiceName
+}
```
Comment (spacewander) [inline internal/core/message/message.go:48]: Need a blank line between functions.

File: internal/core/message/message.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+func (msg *Message) DiscoveryArgs() map[string]string {
+	up := msg.a6Conf.Upstream
+	if up.DiscoveryArgs == nil {
+		return nil
+	}
+	return map[string]string{
+		"namespace_id": up.DiscoveryArgs.NamespaceID,
+		"group_name":   up.DiscoveryArgs.GroupName,
+	}
+}
```
Comment (spacewander) [inline internal/core/message/message.go:67]: Ditto

File: internal/core/storer/store.go [hunk 5]
```diff
@@ -2,27 +2,24 @@ package storer
...
 	Update(context.Context, string, string) error
-	Watch(context.Context, string) <-chan *StoreEvent
+	Watch(context.Context, string) <-chan []*message.Message
 }
 
 type GenericStoreOption struct {
 	BasePath string
 	Prefix   string
-	ObjType  reflect.Type
+	//ObjType  reflect.Type
```
Comment (spacewander) [inline internal/core/storer/store.go]: We can remove it?

File: internal/discoverer/zookeeper.go [hunk 6]
```diff
@@ -47,37 +48,36 @@ func (zd *ZookeeperDiscoverer) Stop() {
...
 	// TODO:
 	// Zookeeper does not need to introduce parameter information like nacos,
 	// so the Update interface cannot be triggered on the framework.
 	return nil
 }
 
-func (zd *ZookeeperDiscoverer) Watch() chan *comm.Message {
+func (zd *ZookeeperDiscoverer) Delete(msg *message.Message) error {
+	// TODO:
+	return nil
```
Comment (spacewander) [inline internal/discoverer/zookeeper.go]: We need to handle it in `discoverer.GetDiscoverer(delMsg.DiscoveryType()).Delete(delMsg)`?

Comment (soulbird) [reply internal/discoverer/zookeeper.go]: Yes, it is needed. And cc@shuaijinchao ,can you assist me in implementing this function?

**Review by spacewander — APPROVED** [review]

**Pull request merged**